src/main/java/services/RestoreFactory.java

```diff
@@ -351,6 +353,14 @@ private String getSyncToken(String username) {
         return storage.getMetaDataFieldForRecord(users.firstElement(), User.META_SYNC_TOKEN);
     }
 
+    // Device ID for tracking usage in the same way Android uses IMEI
+    private String getSyncDeviceId() {
+        if (asUsername == null) {
+            return DEVICE_ID_SLUG;
+        }
+        return String.format("%s|%s|as|%s", DEVICE_ID_SLUG, username, asUsername);
```

Comment:
Hey guys,

Definitely no preference on user_id v. username. user_id seems like it has advantages for the reason simon outlined. Don't think it will be an issue for enikshay in particular due to how we are doing user management, and they are the only consumers of the behavior the device_id flag is being set for, so happy to move forward without it for now, but agree that semantically it is the right long-term behavior.